src/Utilities/functions.jl

```diff
@@ -1821,9 +1821,9 @@ function count_terms(dimension::I, terms::Vector{T}) where {I,T}
     return counting
 end
 
-convert_approx(::Type{T}, func::T; kws...) where {T} = func
+convert_approx(::Type{T}, func::T; kws...) where {T} = MOIU.canonical(func)
```

Comment:
Why is this needed ?

Reply:
It's needed to eliminate variables. Without it, MOI thinks it has more variables in the VAF then there are following elimination.

Reply:
Without this change, we get the following errors for NormOne bridge tests:
```
julia> include("test/Bridges/Constraint/norm_to_lp.jl")
[ Info: Recompiling stale cache file /home/coey/.julia/compiled/v1.3/MathOptInterface/tyub8.ji for MathOptInterface [b8f27783-ece8-5eb3-8dc8-9495eed66fee]
Test Summary: | Pass  Total
NormInfinity  |  139    139
ConstraintFunction: Test Failed at /home/coey/.julia/dev/MathOptInterface/src/Test/UnitTests/basic_constraint_tests.jl:211
  Expression: variables(func) == variables(constraint_function)
   Evaluated: Set(MathOptInterface.VariableIndex[MathOptInterface.VariableIndex(12345677), MathOptInterface.VariableIndex(12345674), MathOptInterface.VariableIndex(12345676), MathOptInterface.VariableIndex(12345675), MathOptInterface.VariableIndex(12345679)]) == Set(MathOptInterface.VariableIndex[MathOptInterface.VariableIndex(12345677), MathOptInterface.VariableIndex(12345676), MathOptInterface.VariableIndex(12345679)])
Stacktrace:
 [1] macro expansion at /home/coey/.julia/dev/MathOptInterface/src/Test/UnitTests/basic_constraint_tests.jl:211 [inlined]
 [2] macro expansion at /home/coey/julia/usr/share/julia/stdlib/v1.3/Test/src/Test.jl:1114 [inlined]
 [3] macro expansion at /home/coey/.julia/dev/MathOptInterface/src/Test/UnitTests/basic_constraint_tests.jl:209 [inlined]
 [4] macro expansion at /home/coey/julia/usr/share/julia/stdlib/v1.3/Test/src/Test.jl:1114 [inlined]
 [5] #basic_constraint_test_helper#39(::Bool, ::Bool, ::Bool, ::typeof(MathOptInterface.Test.basic_constraint_test_helper), ::MathOptInterface.Bridges.Constraint.SingleBridgeOptimizer{MathOptInterface.Bridges.Constraint.NormOneBridge{Float64,F,G,H} where H where G where F,MathOptInterface.Utilities.MockOptimizer{MathOptInterface.Utilities.Model{Float64}}}, ::MathOptInterface.Test.TestConfig, ::getfield(MathOptInterface.Test, Symbol("##16#17")), ::MathOptInterface.NormOneCone, ::Int64) at /home/coey/.julia/dev/MathOptInterface/src/Test/UnitTests/basic_constraint_tests.jl:196
ConstraintFunction: Test Failed at /home/coey/.julia/dev/MathOptInterface/src/Test/UnitTests/basic_constraint_tests.jl:211
  Expression: variables(func) == variables(constraint_function)
   Evaluated: Set(MathOptInterface.VariableIndex[MathOptInterface.VariableIndex(12345677), MathOptInterface.VariableIndex(12345674), MathOptInterface.VariableIndex(12345676), MathOptInterface.VariableIndex(12345675), MathOptInterface.VariableIndex(12345679)]) == Set(MathOptInterface.VariableIndex[MathOptInterface.VariableIndex(12345677), MathOptInterface.VariableIndex(12345676), MathOptInterface.VariableIndex(12345679)])
Stacktrace:
 [1] macro expansion at /home/coey/.julia/dev/MathOptInterface/src/Test/UnitTests/basic_constraint_tests.jl:211 [inlined]
 [2] macro expansion at /home/coey/julia/usr/share/julia/stdlib/v1.3/Test/src/Test.jl:1114 [inlined]
 [3] macro expansion at /home/coey/.julia/dev/MathOptInterface/src/Test/UnitTests/basic_constraint_tests.jl:209 [inlined]
 [4] macro expansion at /home/coey/julia/usr/share/julia/stdlib/v1.3/Test/src/Test.jl:1114 [inlined]
 [5] #basic_constraint_test_helper#39(::Bool, ::Bool, ::Bool, ::typeof(MathOptInterface.Test.basic_constraint_test_helper), ::MathOptInterface.Bridges.Constraint.SingleBridgeOptimizer{MathOptInterface.Bridges.Constraint.NormOneBridge{Float64,F,G,H} where H where G where F,MathOptInterface.Utilities.MockOptimizer{MathOptInterface.Utilities.Model{Float64}}}, ::MathOptInterface.Test.TestConfig, ::getfield(MathOptInterface.Test, Symbol("##18#19")), ::MathOptInterface.NormOneCone, ::Int64) at /home/coey/.julia/dev/MathOptInterface/src/Test/UnitTests/basic_constraint_tests.jl:196
Test Summary:                                                                    | Pass  Fail  Total
NormOne                                                                          |  146     2    148
  MathOptInterface.VectorOfVariables-MathOptInterface.NormOneCone                |   13           13
  MathOptInterface.VectorAffineFunction{Float64}-MathOptInterface.NormOneCone    |   14     1     15
    add_constraint                                                               |    7     1      8
      ConstraintName                                                             |    3            3
      ConstraintFunction                                                         |    1     1      2
      ConstraintSet                                                              |    1            1
    ListOfConstraintIndices                                                      |    1            1
    add_constraints                                                              |    2            2
    is_valid                                                                     |    1            1
    delete                                                                       |    2            2
  MathOptInterface.VectorQuadraticFunction{Float64}-MathOptInterface.NormOneCone |   14     1     15
    add_constraint                                                               |    7     1      8
      ConstraintName                                                             |    3            3
      ConstraintFunction                                                         |    1     1      2
      ConstraintSet                                                              |    1            1
    ListOfConstraintIndices                                                      |    1            1
    add_constraints                                                              |    2            2
    is_valid                                                                     |    1            1
    delete                                                                       |    2            2
  Test mock model                                                                |   15           15
  Test bridged model                                                             |   12           12
ERROR: LoadError: Some tests did not pass: 146 passed, 2 failed, 0 errored, 0 broken.
in expression starting at /home/coey/.julia/dev/MathOptInterface/test/Bridges/Constraint/norm_to_lp.jl:79
```
there are others ways to fix the issue probably, like canonicalizing in the basic constraint tests before getting the variables. @blegat what is your preferred fix?

Reply:
You can do `removevariable` in the bridge like in:
https://github.com/JuliaOpt/MathOptInterface.jl/blob/0ed068ff8342cf2501305963c22708b0b18a4b80/src/Bridges/Constraint/slack.jl#L98

Reply:
OK fixed